example/ExampleMain.cpp

```diff
@@ -166,12 +172,33 @@ LLVM_DIALECTS_VISITOR_PAYLOAD_PROJECT_FIELD(VisitorNest, inner)
 // i.e. if C++ had a strong enough compile-time evaluation (constexpr), it
 // should be possible to evaluate the initialization entirely at compile-time.
 template <bool rpot> const Visitor<VisitorContainer> &getExampleVisitor() {
+  static const auto complexSet = OpSet::fromOpDescriptions(
+      {OpDescription::fromCoreOp(Instruction::Ret),
+       OpDescription::fromIntrinsic(Intrinsic::umin)});
+
   static const auto visitor =
       VisitorBuilder<VisitorContainer>()
           .nest<VisitorNest>([](VisitorBuilder<VisitorNest> &b) {
             b.add<xd::ReadOp>([](VisitorNest &self, xd::ReadOp &op) {
               *self.out << "visiting ReadOp: " << op << '\n';
             });
+            b.addSet<xd::SetReadOp, xd::SetWriteOp>(
+                [](VisitorNest &self, llvm::Instruction &op) {
+                  if (isa<xd::SetReadOp>(op)) {
+                    *self.out << "visiting SetReadOp (set): " << op << '\n';
+                  } else if (isa<xd::SetWriteOp>(op)) {
+                    *self.out << "visiting SetWriteOp (set): " << op << '\n';
+                  }
+                });
+            b.addSet(complexSet, [](VisitorNest &self, llvm::Instruction &op) {
+              if (op.getOpcode() == Instruction::Ret) {
+                *self.out << "visiting Ret (set): " << op << '\n';
+              } else if (auto *II = dyn_cast<IntrinsicInst>(&op)) {
+                if (II->getIntrinsicID() == Intrinsic::umin) {
+                  *self.out << "visiting umin (set): " << op << '\n';
+                }
```

Comment:
Rephrase the checks here as assertions, since the use of `complexSet` should make these a guarantee. We don't want to accidentally miss a bug here.